server/src/concepts/category/useCases/addNewCategory.js

```diff
@@ -0,0 +1,21 @@
+import createCategory from "../repositories/commands.js";
+import { findCategoryByName } from "../repositories/queries.js";
+
+const addNewCategory = async (data) => {
+  const existingCategory = await findCategoryByName(data.name);
+  if (existingCategory) {
+    throw new Error("Category already exist");
+  }
+  const addCategory = await createCategory({
```

Comment:
You do not have to trigger save here, just trigger this function with our data and return its result. Whole saving logic is inside commands
```suggestion
  return await createCategory(data)
```

You do not need this try block below etc.